### PR TITLE
Unskip `test_standalone` from ci.sh

### PR DESCRIPTION
**Pull request by czgdp1807**
## Why are these changes needed?

Follow up, https://github.com/ray-project/ray/pull/21213

Reference comment, https://github.com/ray-project/ray/pull/21213#issuecomment-999050012

cc: @pcmoritz 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by czgdp1807** [comment]
The tests don't run in Github actions. May be we need to deliberately change something in the tests to make them start.

**Comment by pcmoritz** [comment]
Yes that's a good idea. Just commit some white space change to the `test_standalone` file :)

**Comment by czgdp1807** [comment]
Ah! The windows tests are migrated to buildkite from Github actions. `Run Tests` is commented out in `main.yml`.

File: python/ray/serve/tests/test_standalone.py [hunk 1]
```diff
@@ -145,7 +145,6 @@ def connect_in_deployment(*args):
     assert "deployment-ception" in serve.list_deployments()
 
 
-@pytest.mark.skipif(sys.platform == "win32", reason="Failing on Windows")
```
Comment (czgdp1807) [inline python/ray/serve/tests/test_standalone.py:148]: What's the hardware configuration (number of CPUs, RAM) used by buildkite? @simon-mo

Comment (simon-mo) [reply python/ray/serve/tests/test_standalone.py:148]: 4cpus 8g mem

**Review by simon-mo — APPROVED** [review]
Seems like working on buildkite now!

**Comment by czgdp1807** [comment]
> Seems like working on buildkite now!

Yes. Seems so. Is there any way to run the Windows test again on Buildkite? (Like re-run them).

File: python/ray/serve/tests/test_standalone.py (hunk 1 above)
Comment (czgdp1807) [reply python/ray/serve/tests/test_standalone.py:148]: This test passes in 75 - 80s with 4 CPUs and 5 GB RAM. So, I think we are good with unskipping this test for now. cc: @pcmoritz

**Comment by pcmoritz** [comment]
test_http_root_url looks a bit flaky, do you want to look into that @czgdp1807 before we merge this? Feel free to keep skipping it for this PR and do the de-flaking in a separate PR :)

```




================================== FAILURES ===================================
  | _____________________________ test_http_root_url ______________________________
  |  
  | ray_shutdown = None
  |  
  | def test_http_root_url(ray_shutdown):
  | @serve.deployment
  | def f(_):
  | pass
  |  
  | root_url = "https://my.domain.dev/prefix"
  |  
  | port = new_port()
  | os.environ[SERVE_ROOT_URL_ENV_KEY] = root_url
  | >       serve.start(http_options=dict(port=port))
  |  
  | \\?\C:\Users\ContainerAdministrator\AppData\Local\temp\Bazel.runfiles_aydcswul\runfiles\com_github_ray_project_ray\python\ray\serve\tests\test_standalone.py:286:
  | _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  | c:\install\ray\python\ray\serve\api.py:465: in start
  | ray.get(
  | c:\install\ray\python\ray\_private\client_mode_hook.py:105: in wrapper
  | return func(*args, **kwargs)
  | _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  |  
  | object_refs = [ObjectRef(2b690075612da717ac2df4d8492094ad3b8819650100000001000000)]
  |  
  | @PublicAPI
  | @client_mode_hook(auto_init=True)
  | def get(object_refs: Union[ray.ObjectRef, List[ray.ObjectRef]],
  | *,
  | timeout: Optional[float] = None) -> Union[Any, List[Any]]:
  | """Get a remote object or a list of remote objects from the object store.
  |  
  | This method blocks until the object corresponding to the object ref is
  | available in the local object store. If this object is not in the local
  | object store, it will be shipped from an object store that has it (once the
  | object has been created). If object_refs is a list, then the objects
  | corresponding to each object in the list will be returned.
  |  
  | Ordering for an input list of object refs is preserved for each object
  | returned. That is, if an object ref to A precedes an object ref to B in the
  | input list, then A will precede B in the returned list.
  |  
  | This method will issue a warning if it's running inside async context,
  | you can use ``await object_ref`` instead of ``ray.get(object_ref)``. For
  | a list of object refs, you can use ``await asyncio.gather(*object_refs)``.
  |  
  | Args:
  | object_refs: Object ref of the object to get or a list of object refs
  | to get.
  | timeout (Optional[float]): The maximum amount of time in seconds to
  | wait before returning.
  |  
  | Returns:
  | A Python object or a list of Python objects.
  |  
  | Raises:
  | GetTimeoutError: A GetTimeoutError is raised if a timeout is set and
  | the get takes longer than timeout to return.
  | Exception: An exception is raised if the task that created the object
  | or that created one of the objects raised an exception.
  | """
  | worker = global_worker
  | worker.check_connected()
  |  
  | if hasattr(
  | worker,
  | "core_worker") and worker.core_worker.current_actor_is_asyncio():
  | global blocking_get_inside_async_warned
  | if not blocking_get_inside_async_warned:
  | logger.warning("Using blocking ray.get inside async actor. "
  | "This blocks the event loop. Please use `await` "
  | "on object ref with asyncio.gather if you want to "
  | "yield execution to the event loop instead.")
  | blocking_get_inside_async_warned = True
  |  
  | with profiling.profile("ray.get"):
  | is_individual_id = isinstance(object_refs, ray.ObjectRef)
  | if is_individual_id:
  | object_refs = [object_refs]
  |  
  | if not isinstance(object_refs, list):
  | raise ValueError("'object_refs' must either be an object ref "
  | "or a list of object refs.")
  |  
  | # TODO(ujvl): Consider how to allow user to retrieve the ready objects.
  | values, debugger_breakpoint = worker.get_objects(
  | object_refs, timeout=timeout)
  | for i, value in enumerate(values):
  | if isinstance(value, RayError):
  | if isinstance(value, ray.exceptions.ObjectLostError):
  | worker.core_worker.dump_object_store_memory_usage()
  | if isinstance(value, RayTaskError):
  | >                       raise value.as_instanceof_cause()
  | E                       ray.exceptions.RayTaskError(ValueError): ray::HTTPProxyActor.ready() (pid=8816, ip=127.0.0.1, repr=<ray.serve.http_proxy.HTTPProxyActor object at 0x000001C06250B1F0>)
  | E                       OSError: [WinError 10013] An attempt was made to access a socket in a way forbidden by its access permissions
  | E
  | E                       During handling of the above exception, another exception occurred:
  | E
  | E                       ray::HTTPProxyActor.ready() (pid=8816, ip=127.0.0.1, repr=<ray.serve.http_proxy.HTTPProxyActor object at 0x000001C06250B1F0>)
  | E                         File "python\ray\_raylet.pyx", line 629, in ray._raylet.execute_task
  | E                         File "python\ray\_raylet.pyx", line 670, in ray._raylet.execute_task
  | E                         File "python\ray\_raylet.pyx", line 636, in ray._raylet.execute_task
  | E                         File "python\ray\_raylet.pyx", line 640, in ray._raylet.execute_task
  | E                         File "python\ray\_raylet.pyx", line 584, in ray._raylet.execute_task.function_executor
  | E                         File "python\ray\_raylet.pyx", line 2074, in ray._raylet.CoreWorker.run_async_func_in_event_loop
  | E                         File "C:\Miniconda3\lib\concurrent\futures\_base.py", line 437, in result
  | E                           return self.__get_result()
  | E                         File "C:\Miniconda3\lib\concurrent\futures\_base.py", line 389, in __get_result
  | E                           raise self._exception
  | E                         File "c:\install\ray\python\ray\util\tracing\tracing_helper.py", line 483, in _resume_span
  | E                           return await method(self, *_args, **_kwargs)
  | E                         File "c:\install\ray\python\ray\serve\http_proxy.py", line 329, in ready
  | E                           return await done_set.pop()
  | E                         File "c:\install\ray\python\ray\util\tracing\tracing_helper.py", line 483, in _resume_span
  | E                           return await method(self, *_args, **_kwargs)
  | E                         File "c:\install\ray\python\ray\serve\http_proxy.py", line 348, in run
  | E                           raise ValueError(
  | E                       ValueError: Failed to bind Ray Serve HTTP proxy to '127.0.0.1:13474'.
  | E                       Please make sure your http-host and http-port are specified correctly.
  |  
  | c:\install\ray\python\ray\worker.py:1733: RayTaskError(ValueError)
  | ============================== warnings summary ===============================
 

<br class="Apple-interchange-newline">
```

**Comment by czgdp1807** [comment]
I see. It never fails on my Azure VM. I ran it like several times using pytest-repeat and it passed all the times. I will see if there’s a way to deliberately make this fail.

**Comment by czgdp1807** [comment]
From the logs it seems like an access permission problem on CI.

**Comment by czgdp1807** [comment]
I did some investigation and there might be a couple of reasons for this failure,

1. `13474` (the port to which binding is failed) requires admin access for binding. Now, in my Azure VM Anaconda only runs with Administrator access so it can bind to any port. I can't remove that administrator access somehow from Anaconda. However, it is also interesting to note that only port numbers < 1024 require admin access on Windows. **13474 > 1024 so it should work without any special permissions.**
2. `13474` is already in use somewhere on the machine but `new_port` somehow gives that answer and hence the binding fails. We have encountered this kind of issue while dealing with initialising multiple instances  of ray in same Windows machine.

To me 2nd point seems like a more probable case. Anyways, I have skipped the test for now and will look at it later.

**Pull request merged**